### ISSUE TITLE
Plex Music: Fanart for tracks is off

### ISSUE THREAD

**Issue by croneter** [croneter/PlexKodiConnect#757]
> I have a question regarding fanart for music. But I‘m not sure whether this is a problem of PKC or not.

> I have a huge library of Soundtracks in Plex. Each artist has a poster and a background (showing the artist). And also each album has a poster (front cover) and a background (movie wallpaper). All are synced via PKC to Kodi (Intel NUC with LibreELEC). When I‘m browsing through the list of artists, each artist is visualised and the belonging background is shown when the artist is highlighted. After selecting an artist the associated albums are shown and when I browse through the list of albums, all belonging backgrounds (movie wallpaper) are displayed.

> And now comes the strange thing: when I select an album and the track list is shown, for some albums the belonging movie wallpaper is visible (as expected), but for tracks of other albums the artist background or a random background is shown. It seems that these tracks are incident with the artist background and not with the album background.
After further investigations I‘m pretty sure that all albums are affected of all artists with a beginning letter from A to J.

> I have another Kodi/PKC player on my windows pc and all albums are affected there.

> This behavior occurs with different skins. So my thought was that maybe the way how PKC syncs can be the problem. Or do I miss something?

**Comment by Violetto** [comment]
After some investigations I have more information about this misbehavior. 

To bring back all the missing album backrounds on track level I have to reset the library; repairing does not solve the issue. During the first run after resetting everything is fine. All album backgrounds are available when belonging tracks are selected or played. 

The problem starts with the first restart of Kodi. During the first minute after restart, a couple (maybe three or four) of album backgrounds are gone, not on album level, but on track level. It begins with the first artist in alphabetical order (here: Alan Menken). All four albums of Alan Menken in my library have lost their album background on track level and the artist background is shown instead.

With the next Kodi restart albums of the next artist (here: Alan Silvestri) are affected. I have more albums from this artist in my library, so I'm able to observe more exactly whats going on. Also on album level the first items in alphabetical order are affected, starting with "A Christmas Carol", the soundtracks from the "Back to the Future" movies, "Beowulf" and so on. With each Kodi restart more albums show this misbehavior. Maybe this is a kind of sync issues.

During my investigations I observed a worse issue: Whole albums are "erased" from library from one moment to another when I was in the same menu to enter album track list and back to show the albums. They wasn't listed anymore under the artist. When using the search function, these albums are listed in the search result with album cover, but when I select this search list item (to list the tracks), the system says "No results found". These albums cannot be played anymore. Because this issue occurs at the same time with the background issue, both may have the same root cause.

Hope this helps.

**Comment by croneter** [comment]
@Violetto Thanks a ton for these insights. However, I cannot replicate this issue. Are you using addon paths or direct paths?

Could you be so kind and invite me as a friend for you PMS so I could solve this issue using your library?

**Comment by Violetto** [comment]
@croneter Invitation is out. I'm using direct paths. 
Thank you for your help!

**Comment by croneter** [comment]
@Violetto Thanks! But I think you need to switch on sharing of your music library for me

**Comment by Violetto** [comment]
@croneter Hmm, I'm not familiar with the settings. 
For your account "Share > All Libraries" is set. 
Under restrictions now I have set "Allow Sync" and "Allow Plugins" too. 
For "Albums > Labels" "Allow all labels" is shown.
Is there anything else to set?

**Comment by croneter** [comment]
Don't know, but not a single library is visible. Please see here: https://support.plex.tv/articles/204232573-restricting-the-shares/

**Comment by Violetto** [comment]
Next try. Instead of "All Libraries" I share only the Soundtrack library now.
I restarted the plex server additionaly.

**Comment by croneter** [comment]
That worked, thanks!

**Comment by croneter** [comment]
I still cannot replicate your issue. What skin are you using? Do albums of another artist disappear if you start a manual library sync? (Addons - Video addons - Plexkodiconnect)

**Comment by croneter** [comment]
Please also share a debug log after a reboot / a manual sync. I might see something in there

**Comment by Violetto** [comment]
_What skin are you using?_
Usually I use the "Aeon Nox: Silvo" skin, but the issue occurs with the default skin too (see below).

_Do albums **of another artist** disappear if you start a manual library sync?_
Not shure if I understand this question right, but I can confirm, that starting a manual library sync is enough to lead to the described issues (loss of fanart for tracks and disappering albums).

To reproduce this misbehavior I deleted my Windows Kodi installation and started from scratch with the Kodi v18.4 "Leia" 64bit installer. 
Then I installed PKC version 2.9.9, customized the music paths for Direct Paths and after restarting Kodi I selected only the Soundtrack Library for sync.

After the sync was completed I checked all 4 albums of the artist "Alan Menken" and the album fanart was visible in tracks as it should be (example "Aladdin").

![screenshot002](https://user-images.githubusercontent.com/26852962/66335039-7ada1b80-e93a-11e9-8ac7-31e6bc0c9e3e.png)

Maybe I restarted Kodi one or two times after that.

Then I started a manual library sync and checked the albums again. After the first sync I saw no difference: All 4 albums were available and the tracks had the right album fanarts.

After the next manual library sync the tracks of all albums from Alan Menken had lost their fanart (instead the artist fanart was shown, example "Aladdin") AND the album "Beauty and the Beast" had no tracks anymore. 

![screenshot006](https://user-images.githubusercontent.com/26852962/66335058-83caed00-e93a-11e9-8fc0-6fc80fb248e5.png)

![screenshot007](https://user-images.githubusercontent.com/26852962/66335086-92b19f80-e93a-11e9-8f76-7fb964dd1979.png)

Here is the regarding logfile:
[kodi.log](https://github.com/croneter/PlexKodiConnect/files/3698779/kodi.log)

I can reproduce this issue all the time. It only differs in the count of affected albums after a sync (regarding the loss of fanart) and when or which album disappears.

Let me know if I can do more to achieve a solution.
And again: Thanks for your support!

**Comment by croneter** [comment]
I haven't managed to reproduce your issue yet, even using your PMS 😢 
I do find these lines in your log, however:
```
2019-10-07 18:33:43.415 T:6932   DEBUG: MUSIC_INFO::CMusicInfoScanner::Process - Starting scan
2019-10-07 18:33:43.445 T:6932   DEBUG: MUSIC_INFO::CMusicInfoScanner::DoScan Skipping dir 'smb://DISKSTATION/usbshare2/Soundtracks (Lossless FLAC)/Abigail Mead/Full Metal Jacket (Warner Bros - 1987)/' due to no change
2019-10-07 18:33:43.467 T:6932   DEBUG: MUSIC_INFO::CMusicInfoScanner::DoScan Skipping dir 'smb://DISKSTATION/usbshare2/Soundtracks (Lossless FLAC)/Adam Fields/Dawson's Creek (Promo - 2003)/' due to no change
2019-10-07 18:33:43.503 T:6932   DEBUG: MUSIC_INFO::CMusicInfoScanner::DoScan Rescanning dir 'smb://DISKSTATION/usbshare2/Soundtracks (Lossless FLAC)/Akira Ifukube/Godzilla (50th Anniversary Edition - La-La Land - 2004)/' due to change

2019-10-07 18:33:43.577 T:6932   DEBUG: unrecognized XipComment name: ADDED_TIMESTAMP
2019-10-07 18:33:43.577 T:6932   DEBUG: unrecognized XipComment name: BUYCDURL
2019-10-07 18:33:43.577 T:6932   DEBUG: unrecognized XipComment name: DESCRIPTION
2019-10-07 18:33:43.577 T:6932   DEBUG: unrecognized XipComment name: ORGANIZATION
2019-10-07 18:33:43.577 T:6932   DEBUG: unrecognized XipComment name: TRACKTOTAL

2019-10-07 18:33:47.067 T:6932   DEBUG: SELECT albumview.*,albumartistview.*  FROM albumview  JOIN albumartistview ON albumview.idAlbum = albumartistview.idAlbum  WHERE albumview.idAlbum = 1294  ORDER BY albumartistview.iOrder
2019-10-07 18:33:47.135 T:6932   DEBUG: MUSIC_INFO::CMusicInfoScanner::DoScan Rescanning dir 'smb://DISKSTATION/usbshare2/Soundtracks (Lossless FLAC)/Alan Menken/Aladdin (Walt Disney - 1992)/' due to change
2019-10-07 18:33:47.176 T:6932   ERROR: Could not find 'smb://DISKSTATION/usbshare2/Soundtracks (Lossless FLAC)/Alan Menken/Aladdin (Walt Disney - 1992)/01 - Bruce Adler , Arabian Nights.wav' referenced in cue, case sensitivity issue?
2019-10-07 18:33:47.178 T:6932   ERROR: Could not find 'smb://DISKSTATION/usbshare2/Soundtracks (Lossless FLAC)/Alan Menken/Aladdin (Walt Disney - 1992)/02 - Robin Williams , Legend of the Lamp.wav' referenced in cue, case sensitivity issue?
2019-10-07 18:33:47.180 T:6932   ERROR: Could not find 'smb://DISKSTATION/usbshare2/Soundtracks (Lossless FLAC)/Alan Menken/Aladdin (Walt Disney - 1992)/03 - Brad Kane 'Aladdin' , One Jump Ahead.wav' referenced in cue, case sensitivity issue?
2019-10-07 18:33:47.180 T:6932   ERROR: Track split over multiple files, unsupported.
```
It looks like Kodi scans your music, even though PKC should have disabled that for the direct paths in question! Reason is that Plex supplies everything Kodi needs - Kodi trying to scan files and folders just messes things up like in your case. 
PKC changes your `advancedsettings.xml` to include the following:
```
<audio>
<excludefromscan>
  <regexp>^/volumeUSB2/usbshare/Soundtracks (Lossless FLAC)/</regexp>
  <regexp>^/volumeUSB2/usbshare/Soundtracks (MP3)/</regexp>
</excludefromscan>
</audio>
```
That regex there seems to be off, because it does NOT match your paths:
```
2019-11-14 14:57:30.930 T:3820 WARNING: MUSIC_INFO::CMusicInfoScanner::Process directory '/volumeUSB2/usbshare/Soundtracks (Lossless FLAC)/Hans Zimmer/The Da Vinci Code (Decca - 2006)/' does not exist - skipping scan.
```

**Comment by croneter** [comment]
Got it! The parenthesis `()` are the issue - they're not escaped for regex matching. See https://stackoverflow.com/questions/4202538/escape-regex-special-characters-in-a-python-string

**Comment by croneter** [comment]
Fixed in PKC **BETA** version 2.10.3!! 💪 😆
